cmd/orch-tester/orch_tester.go

```diff
@@ -68,6 +70,134 @@ type broadcasterConfig struct {
 	maxPricePerUnit *int
 }
 
+type broadcasterMetrics struct {
+	// Distribution metrics grouped by metric name
+	lastSum   map[string]float64
+	lastCount map[string]int
+	rateSum   map[string]float64
+	rateCount map[string]int
+
+	// Error count metrics by grouped by error_code tag
+	lastErrs map[string]int
+	incErrs  map[string]int
+
+	mu sync.Mutex
+}
+
+func newBroadcasterMetrics() *broadcasterMetrics {
+	return &broadcasterMetrics{
+		lastSum:   map[string]float64{},
+		lastCount: map[string]int{},
+		rateSum:   map[string]float64{},
+		rateCount: map[string]int{},
+		lastErrs:  map[string]int{},
+		incErrs:   map[string]int{},
+		mu:        sync.Mutex{},
+	}
+}
+
+func (bm *broadcasterMetrics) reset() {
+	bm.mu.Lock()
+	defer bm.mu.Unlock()
+
+	bm.rateSum = map[string]float64{}
+	bm.rateCount = map[string]int{}
+	bm.incErrs = map[string]int{}
```

Comment:
Don't you need to reset the other fields here as well?

Reply:
Also, if we do reset everything here like this, we could even make the constructor call this with something like
```
bm := &broadcasterMetrics{}
bm.reset()
return bm
```
WDYT?

Reply:
No, we don't reset all the fields, only the rate fields. The other fields (`lastSum`, `lastCount`, `lastErrs`) always keep the latest accumulated value from exported metrics. That is because census always aggregate metrics, so we need to know the last measurement. Metrics are tricky....